nifi-nar-bundles/nifi-hive-bundle/nifi-hive_1_1-processors/src/main/java/org/apache/nifi/processors/hive/UpdateHive_1_1Table.java

```diff
@@ -562,7 +686,35 @@ private synchronized void checkAndUpdateTableSchema(final ProcessSession session
                 outputPath = tableLocation + "/" + String.join("/", partitionColumnsLocationList);
             }
 
-            session.putAttribute(flowFile, ATTR_OUTPUT_PATH, outputPath);
+            // If updating field names, return a new RecordSchema, otherwise return null
+            OutputMetadataHolder outputMetadataHolder;
+            if (!tableCreated && updateFieldNames) {
+                List<RecordField> inputRecordFields = schema.getFields();
+                List<RecordField> outputRecordFields = new ArrayList<>();
+                Map<String,String> fieldMap = new HashMap<>();
+
+                for (RecordField inputRecordField : inputRecordFields) {
+                    final String inputRecordFieldName = inputRecordField.getFieldName();
+                    boolean found = false;
+                    for (String hiveColumnName : hiveColumns) {
+                        if (inputRecordFieldName.equalsIgnoreCase(hiveColumnName)) {
```

Comment:
If 'Update Fields Names' has been set to true, it always remap the records. Even if all the columns names are identical and the conversion would not be needed. It could be optimised to convert the records only if necessary.

In that case the record reader and writer must have the same data format because some FFs will be converted while others will not. Actually, it is already a requirement for a create + update scenario (create never converts) and should be documented for the users.

Reply:
Agreed, I will change the behavior and add the appropriate documentation.